rust/noosphere-ns/src/dht/swarm.rs

```diff
@@ -83,8 +82,20 @@ impl DhtBehavior {
             )));
 
             // TODO(#99): Use SphereFS storage
-            let store = kad::record::store::MemoryStore::new(local_peer_id.to_owned());
-            Kademlia::with_config(local_peer_id.to_owned(), store, cfg)
+            let store = kad::store::MemoryStore::new(local_peer_id.to_owned());
+            let mut behaviour =
+                KademliaBehaviour::with_config(local_peer_id.to_owned(), store, cfg);
+
+            // Updating to libp2p 0.53 introduced DHT nodes adjusting their modes dynamically,
+            // where if a node does not have an external address, it switches to `Mode::Client`.
+            // This improves the network by culling non-accessible nodes.
+            // Manually setting to `Mode::Server` mode prevents this dynamic switching.
+            // We could explore this optimization in the future, but for now,
+            // we need to run on nightly rust, and some challenges with external addresses
+            // in tests.
+            // https://github.com/libp2p/rust-libp2p/pull/4503
```

Comment:
Can we file an issue for "explore this optimization in the future"?

Reply:
#814